src/module-elasticsuite-catalog/Helper/AbstractAttribute.php

```diff
@@ -172,7 +172,7 @@ public function prepareIndexValue(AttributeInterface $attribute, $storeId, $valu
         }
 
         $value = array_map($this->attributeMappers[$mapperKey], $value);
-        $value = array_filter($value);
+        $value = array_diff(array_map('trim', $value), ['', null, false]);
```

Comment:
A bit ugly, I understand why you removed the `boolval` later, mostly because `trim(true)` would result in `"1"` and `trim(false)` would result in `''`.